snap/integrity/dm_verity/veritysetup.go

```diff
@@ -0,0 +1,106 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package dm_verity
+
+import (
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+type VeritySuperBlock struct {
```

Comment:
Are we sure that the output of `veritysetup format` is all that we need? I see some differences between this struct and the superblock as described as `struct verity_sb` in https://gitlab.com/cryptsetup/cryptsetup/-/wikis/DMVerity.

Also I worry a bit about relying in the output of a tool as it might change in the future - maybe it would make sense to create the full verity file and then take from there the superblock.

Reply:
The differences are the signature which we don't need and the 2 pads which are used to pad the C struct. I agree with the approach to get it from the generated superblock instead of the cmdline tool output though, seems more robust.